src/DIRAC/FrameworkSystem/Service/TokenManagerHandler.py

```diff
@@ -211,39 +211,40 @@ def export_getToken(
             return result
         idpObj = result["Value"]
 
-        if userGroup and (result := idpObj.getGroupScopes(userGroup))["OK"]:
+        if userGroup and (result := idpObj.getGroupScopes(userGroup)):
             # What scope correspond to the requested group?
-            scope = list(set((scope or []) + result["Value"]))
+            scope = list(set((scope or []) + result))
 
         # Set the scope
         idpObj.scope = " ".join(scope)
 
-        # Let's check if there are corresponding tokens in the cache
-        cacheKey = (username, idpObj.scope, audience, identityProvider)
+        # Let's check if there is a corresponding token in the cache
+        cacheKey = (userName, idpObj.scope, audience, identityProvider)
         if self.__tokensCache.exists(cacheKey, requiredTimeLeft):
             # Well we have a fresh record containing a Token object
             token = self.__tokensCache.get(cacheKey)
             # Let's check if the access token is fresh
             if not token.is_expired(requiredTimeLeft):
                 return S_OK(token)
-            # It seems that it is no longer valid for us, but whether there is a refresh token?
-            if token.get("refresh_token"):
-                # Okay, so we can try to refresh tokens
-                if (result := idpObj.refreshToken(token["refresh_token"]))["OK"]:
-                    # caching new tokens
-                    self.__tokensCache.add(
-                        cacheKey,
-                        result["Value"].get_claim("exp", "refresh_token") or self.DEFAULT_RT_EXPIRATION_TIME,
-                        result["Value"],
-                    )
-                    return result
-                self.log.verbose(f"Failed to get token with cached tokens: {result['Message']}")
-                # Let's try to revoke broken token
-                idpObj.revokeToken(token["refresh_token"])
 
         err = []
-        # The cache did not help, so let's make an exchange token
-        result = Registry.getDNForUsername(username)
+        # Do we have an oidc-agent configured ? If yes, try to use it
+        if userName in idpObj.parameters.get("OIDCAgentUsers", []):
```

Comment:
The `OIDC-agent` will not be supported within DIRAC, therefore I would not include it as an option of the `TokenManagerHandler` (IMHO, it adds more complexity than before where the implementation was encapsulated in a separate class).

Last time you said that it was a recommended solution, where did you hear/see this?
I still don't get why we need it somewhere in the code.

Reply:
OK, I will drop it. After the initial testing, this is not actually important.